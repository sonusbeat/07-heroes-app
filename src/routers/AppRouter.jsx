import { useContext } from 'react';
import { BrowserRouter as Router, Switch } from "react-router-dom";

import AuthContext from '../auth/AuthContext';
import PrivateRoute from "./PrivateRoute";
import LoginScreen from '../components/login/LoginScreen';
import DashboardRoutes from './DashboardRoutes';
import PublicRoute from './PublicRoute';


const AppRouter = () => {
  const { user:{ logged } } = useContext(AuthContext);

  return (
    <Router>
      <div>
        <Switch>

          <PublicRoute
            exact
            path="/login"
            isAuthenticated={ logged }
            component={ LoginScreen }
          />

          <PrivateRoute
            path="/"
            component={ DashboardRoutes }
            isAuthenticated={ logged }
          />

        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
