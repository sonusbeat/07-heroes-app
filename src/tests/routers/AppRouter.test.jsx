import { mount } from 'enzyme';
import AuthContext from '../../auth/AuthContext';
import AppRouter from '../../routers/AppRouter';

describe("Pruebas en <AppRouter />", () => {

  test("Debería de mostrar de mostrar login si no esta autenticado", () => {

    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged: false
      }
    };

    const wrapper = mount(
      <AuthContext.Provider value={ contextValue }>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Debería de mostrar el componente marvel si está autenticado", () => {

    const contextValue = {
      dispatch: jest.fn(),
      user: {
        name: "Daniel",
        logged: true
      }
    };

    const wrapper = mount(
      <AuthContext.Provider value={ contextValue }>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper.find(".navbar").exists()).toBe( true );

  });

});
