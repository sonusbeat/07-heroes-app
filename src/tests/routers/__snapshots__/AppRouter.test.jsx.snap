// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Deber√≠a de mostrar de mostrar login si no esta autenticado 1`] = `
<div>
  <section
    className="container mt-5"
  >
    <h2>
      Login
    </h2>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Acceder
    </button>
  </section>
</div>
`;
